src/main/kotlin/de/bund/digitalservice/useid/tracking/matomo/MatomoTrackingService.kt

```diff
@@ -20,26 +21,43 @@ import kotlin.text.Charsets.UTF_8
  * Documentation about matomo events
  * https://matomo.org/guide/reports/event-tracking/
  */
-@Profile("!local")
 @Service
 class MatomoTrackingService(trackingProperties: TrackingProperties, private val webRequests: WebRequests) {
 
     private val log = KotlinLogging.logger {}
     private val siteId = trackingProperties.matomo.siteId
     private val domain = trackingProperties.matomo.domain
+    private val dimensionIdTenant = trackingProperties.matomo.dimensionIdTenant
 
-    fun constructEventURL(e: MatomoEvent): String {
-        val session = e.sessionId?.let { "&uid=$it" } ?: ""
-        val userAgent = e.userAgent?.let { "&ua=${UriUtils.encode(e.userAgent, UTF_8)}" } ?: ""
-        val url = "$domain?idsite=$siteId&rec=1&ca=1&e_c=${e.category}&e_a=${e.action}&e_n=${e.name}$session$userAgent"
-        log.debug { url }
-        return url
+    fun constructEventUrl(e: MatomoEvent): String {
+        return UriComponentsBuilder
+            .newInstance()
+            .scheme("https")
+            .host(domain)
+            .queryParam("idsite", siteId)
+            .queryParam("rec", 1)
+            .queryParam("ca", 1)
+            .queryParamIfPresent("e_c", enc(e.category))
+            .queryParamIfPresent("e_a", enc(e.action))
+            .queryParamIfPresent("e_n", enc(e.name))
+            .queryParamIfPresent("uid", enc(e.sessionId))
+            .queryParamIfPresent("ua", enc(e.userAgent))
+            // custom dimension for tenantId -> https://matomo.org/faq/reporting-tools/create-track-and-manage-custom-dimensions/
+            .queryParamIfPresent("dimension$dimensionIdTenant", enc(e.tenantId))
+            .build()
+            .toString()
+    }
+
+    // UriComponentBuilder does not support query param encoding
+    private fun enc(input: String?): Optional<String> {
```

Comment:
```suggestion
    private fun uriEncode(input: String?): Optional<String> {
```

Idiomatic change. `enc` imo, is short for encryption, not encoding.